solarnet/README.md

```diff
@@ -93,4 +123,73 @@ one-liners be automated.
 
 ### unsupported parameter for module: restart_policy
 
-You're probably using the system version of Ansible, and it is outdated (< 1.9). Make sure to follow the Ansible setup steps above, and load the virtualenv. This will load a working version of Ansible.
+You're probably using the system version of Ansible, and it is outdated (< 1.9).
+Make sure to follow the Ansible setup steps above, and load the virtualenv.
+This will load a working version of Ansible.
+
+## Monitoring
+
+We use Prometheus to scrape and store timeseries
+from IPFS and the hosts themselves.
+Grafana provides the dashboard UI.
+
+Both are available at http://metrics.i.ipfs.io
+http://metrics.i.ipfs.io/prometheus, respectively.
+
+- You need to be peered into the Hyperboria cjdns network
+  in order to reach the address that this domain name points to.
+- Your cjdns address needs to be whitelisted.
+
+### Cjdns
+
+In addition to serving http://h.gateway.ipfs.io,
+we use cjdns for a very simple VPN based on an IP address whitelist.
+
+```sh
+$ git clone https://github.com/hyperboria/cjdns.git
+$ cd cjdns/
+$ ./do
+$ ./cjdroute --genconf > cjdroute.conf
+```
+
+This creates a `tunX` network interface,
+which grabs all `fc00::/8` traffic and hands it to cjdns.
+
+Scripts for various init systems are provided in `contrib/`.
+
+### Peering
+
+Cjdns nodes peer automatically on local networks,
+but on WANs like the internet, peering requires credentials.
+You can peer with the Solarnet cjdns nodes
+by using one of the `cjdns_authorized_passwords` in `secrets.yml`.
+
+Note that peering with any other node in the Hyperboria network is sufficient, too.
+
+You can check the status of your peerings:
+
+```sh
+$ watch tools/peerStats
+```
+
+Or check connectivity to solarnet:
+
+```sh
+$ ping6 h.gateway.ipfs.io
+```
+
+### Authentication
+
+Since IP packets in cjdns are authenticated and encrypted,
+we can use the HTTP client's IP address for authentication,
+instead of Basic Auth or another login mechanism.
```

Comment:
Thanks this sounds right to me. (Nonce per packet with a counter). I'll look at the spec more closely and compare to how DTLS works



—
Sent from Mailbox

On Tue, Jul 14, 2015 at 6:56 AM, Lars Gierth <notifications@github.com>
wrote:

>> +
>> +```sh
>> +$ watch tools/peerStats
>> +```
>> +
>> +Or check connectivity to solarnet:
>> +
>> +```sh
>> +$ ping6 h.gateway.ipfs.io
>> +```
>> +
>> +### Authentication
>> +
>> +Since IP packets in cjdns are authenticated and encrypted,
>> +we can use the HTTP client's IP address for authentication,
>> +instead of Basic Auth or another login mechanism.
> There's a go implementation of this btw: https://github.com/nsjph/cryptoauth
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/ipfs/infrastructure/pull/44/files#r34569931